vmware-event-router/hack/run_integration_tests.sh

```diff
@@ -0,0 +1,130 @@
+KIND_WAIT=2m    # wait for control plane to show ready
+KIND_CLUSTER="veba-integration"
+# using custom image built with kind since not all versions are pushed to kindtest
+KIND_IMAGE="embano1/node:${K8S_VERSION}"
+#################################
+
+tmp_dir=$(mktemp -d -t ci-XXXXXXXXXX)
+
+cecho() {
+    local default_msg="No message passed."
+    message=${1:-$default_msg}
+    color=${2:-$RESET}
+
+    counter=$((${counter:-0} + 1))
+    echo -e ${color}"\n"\[ ${counter} \] ${message} ${RESET}
+    return
+}
+
+function cleanup() {
+    # capture previous return val, e.g. error
+    rv=$?
+    # don't run cleanup in CI env, e.g. Github Action sets CI=true
+    if [ "${CI:-no}" != true ]; then
+        cecho "Running cleanup and then exiting with code ${rv}" ${GREEN}
+        rm -rf ${tmp_dir} && rm -rf faas-cli-* && rm -rf template && ${KINDBIN} delete cluster --name ${KIND_CLUSTER}
+    fi
+    exit $rv
+}
+trap cleanup INT TERM EXIT
+
+# create kind cluster
+cecho "---> Creating Kubernetes cluster (${K8S_VERSION}) with max wait time: ${KIND_TIMEOUT}" ${GREEN}
+${TIMEOUT_CMD} ${KIND_TIMEOUT} ${KINDBIN} create cluster --name ${KIND_CLUSTER} --wait ${KIND_WAIT} --image ${KIND_IMAGE}
+
+# generate password used by OpenFaaS for basic_auth
+export OF_PASSWORD=$(head -c 12 /dev/urandom | shasum | cut -d' ' -f1)
+
+# if running as Github Actions make it available to other steps
+if [ "${CI:-no}" = true ]; then
+    echo "::set-env name=OF_PASSWORD::$OF_PASSWORD"
+fi
+
+# deploy faas-netes
+cecho "---> Deploying OpenFaaS (${OF_VERSION})" ${GREEN}
+git clone https://github.com/openfaas/faas-netes ${tmp_dir} && git -C ${tmp_dir} checkout ${OF_VERSION}
+kubectl create -f ${tmp_dir}/namespaces.yml
+kubectl -n openfaas create secret generic basic-auth --from-literal=basic-auth-user=admin --from-literal=basic-auth-password="$OF_PASSWORD"
+kubectl create -f ${tmp_dir}/yaml
+
+cecho "---> Waiting up to ${OF_TIMEOUT} for OpenFaaS gateway to become ready" ${GREEN}
+${TIMEOUT_CMD} ${OF_TIMEOUT} /bin/bash -c 'while [[ $READY -lt 1 ]]; do READY=$(kubectl -n openfaas get deploy gateway -o json | '"${JQBIN}"' .status.readyReplicas); sleep 1; done'
+
+# start kubectl port-forwarding
+cecho "---> Starting port-forwarding to OpenFaaS gateway" ${GREEN}
+kubectl -n openfaas port-forward ${PORT_FWD} >/dev/null &
+
+# get faas-cli
+cecho "---> Downloading faas-cli (${CLI_VERSION})" ${GREEN}
+
+case "${OSTYPE}" in
+darwin*) FAAS_CLI="faas-cli-darwin" ;;
+msys*) FAAS_CLI="faas-cli.exe" ;;
+*) FAAS_CLI="faas-cli" ;;
+esac
+
+curl --silent --show-error --fail -LO https://github.com/openfaas/faas-cli/releases/download/${CLI_VERSION}/${FAAS_CLI}
+chmod +x ./${FAAS_CLI}
+echo ${OF_PASSWORD} | ./${FAAS_CLI} login --password-stdin
+
+# deploy functions used by integration tests
+cecho "---> Downloading templates and installing functions" ${GREEN}
+./${FAAS_CLI} template pull
+./${FAAS_CLI} template pull --overwrite https://github.com/openfaas-incubator/python-flask-template
```

Comment:
Do we use the classic templates? If not we can directly merge those two pulls with `faas template store pull python3-flask` which essentially contains the `python-flask-template`

Reply:
`of-echo` still uses the classic template (without status conditions), but could be changed in another PR so we can collapse and speed up the template pull.